File: 5_longest_palindromic_substring/longest_palindromic_substring.go
func longestPalindrome(s string) string {
	// 0.
	result := ""
	// 1.
	for i := 0; i < len(s); i++ {
		// 2.
		odd := helper(s, i, i)
		// 3.
		if len(odd) > len(result) {
			result = odd
		}
		// 4.
		even := helper(s, i, i+1)
		// 5.
		if len(even) > len(result) {
			result = even
		}
	}

	// 6.
	return result
}

func helper(s string, l, r int) string {
	// 0.
	for l >= 0 && r < len(s) && s[l] == s[r] {
		// 1.
		l--
		r++
	}

	// 2.
	return string(s[l+1 : r])
}

File: 5_longest_palindromic_substring/longest_palindromic_substring.js
// Approach 1
/**
 * @param {string} s
 * @return {string}
 */
var longestPalindrome = function (s) {
  // 0.
  var result = "";
  // 1.
  for (let i = 0; i < s.length; i++) {
    // 2.
    var odd = expander(s, i, i);
    // 3.
    if (odd.length > result.length) {
      result = odd;
    }
    // 4.
    var even = expander(s, i, i + 1);
    // 5.
    if (even.length > result.length) {
      result = even;
    }
  }

  // 6.
  return result;
};

function expander(s, l, r) {
  while (l >= 0 && r < s.length && s[l] === s[r]) {
    l -= 1;
    r += 1;
  }
  return s.slice(l + 1, r);
}
